### DISCUSSION TITLE
Refetch timing for invalidateQueries

### DISCUSSION THREAD

**Discussion by himorishige** [TanStack/query#4543] [Q&A] [answered]
The default options are as follows:

```
refetch: false
staleTime: Infinity
```

For example, if I execute `invalidateQueries(['no1'])` on a `queryKey['no1']` that is inactive
My assumption was that the cache of `queryKey['no1']` would be refetched and updated.
However, in reality, the cache is stale, but it is not refetched and the cache is not updated.

sample codesandbox:
https://codesandbox.io/p/sandbox/mystifying-lederberg-yu145g

Expected behavior:
1. get two queries on the Home screen
2. perform invalidate on About screen
3. refetch the two queries in the background
4. go to the Home screen
5. new cache values are displayed

When I explicitly passed the filters as follows, the refetching works at the same time as invalidate and the cache is updated.
`invalidateQueries(['todo'], { type: 'all' })`

I assumed that the default for filters is `type: 'all'`, but would this be the correct behavior difference?
https://tanstack.com/query/v4/docs/guides/filters

**Comment by TkDodo** [comment, marked as answer]
for `invalidateQueries`, `type` defaults to `'active'`, from [the docs](https://tanstack.com/query/v4/docs/guides/query-invalidation):

> If the query is currently being rendered via useQuery or related hooks, it will also be refetched in the background

**Reply by himorishige** [reply]
Thank you!
I was mistaken about it...
